core/src/main/java/de/bwaldvogel/mongo/oplog/OplogCursor.java

```diff
@@ -38,7 +43,18 @@ public boolean isEmpty() {
         return documents;
     }
 
-    OplogPosition getPosition() {
+    private void emulateWaitingForAllShards() {
+        try {
+            // artificial delay to avoid 100% CPU usage when starting multiple ChangeStreams
+            // emulates real ChangeStream behaviour of waiting for all shards to provide data
+            TimeUnit.MILLISECONDS.sleep(100);
```

Comment:
I doubt that this kind of artificial delay is a good implementation.

Reply:
what would you suggest as an alternative? the problem we have observed is that without this delay the clients are constantly polling and causing 100% CPU usage. our tests do not complete and time out. adding the delay resolves the issue.

note, the production MongoDB behaviour has a delay/timeout after waiting for updates from other shards.